### PR TITLE
fix: strip @external fields in interface extensions

### PR DESCRIPTION
**Pull request by lennyburdette**
Before this change, including an external field in an interface
extension would fail composition in the gateway with an error:

```
Field "Product.id" already exists in the schema. It cannot also be defined in this type extension.
```

### CODE REVIEW

File: packages/apollo-federation/src/composition/__tests__/utils.test.ts [hunk 1]
```diff
@@ -37,6 +41,8 @@ describe('Composition utility functions', () => {
         type Mutation {
           updateProduct: Product
         }
+
+        extend interface Account @key(fields: "id")
```
Comment (jbaxleyiii) [inline packages/apollo-federation/src/composition/__tests__/utils.test.ts]: @lennyburdette this interface needs the `id` field defined so the query planner can ask for the `{ id }` on that interface

File: packages/apollo-federation/src/composition/utils.ts [hunk 2]
```diff
@@ -91,6 +91,31 @@ export function stripExternalFieldsFromTypeDefs(
         fields,
       };
     },
+    InterfaceTypeExtension(node) {
+      let fields = node.fields;
```
Comment (JakeDawkins) [inline packages/apollo-federation/src/composition/utils.ts]: It looks like this is identical to the `ObjectTypeExtension`. Could you extract the logic into a function and just do

```
ObjectTypeExtension: removeExternalFieldsFromExtensions,
InterfaceTypeExtension: removeExternalFieldsFromExtensions
```

Comment (lennyburdette) [reply packages/apollo-federation/src/composition/utils.ts]: Will do! The type signatures are different so I went with the cheap approach first, but I'll learn me some typescript and make them the same.

Comment (lennyburdette) [reply packages/apollo-federation/src/composition/utils.ts]: Done!

**Pull request merged**